### PR TITLE
watchdog logging and single instance enforcement

### PR DESCRIPTION
**Pull request by zanderz**
Added an advanced flag, `--log-prefix`, so that each watchdog instance gets its own uniquely named log. Also all the logic to upload the last 5 of these.

### CODE REVIEW

File: go/install/install_windows.go [hunk 1]
```diff
@@ -166,6 +166,43 @@ func InstallLogPath() (string, error) {
...
+	}
+
+	if len(watchdogLogFiles) == 0 {
+		fmt.Fprintf(logFile, "   --- NO WATCHDOG LOGS FOUND!?! ---\n")
+	}
+	for _, path := range watchdogLogFiles {
+		fmt.Fprintf(logFile, "   --- %s ---\n", path)
+
+		// append the files
+		logContent, err := ioutil.ReadFile(path)
```
Comment (akalin-keybase) [inline go/install/install_windows.go]: seems cleaner to do

```
func() {
  fd, ... := os.Open(path)
  defer fd.Close()
  ... := io.Copy(logContent, fd)
  ...
}()
```

**Comment by zanderz** [comment]
thanks, Fred, I double checked it still works.

File: go/libkb/globals.go [hunk 2]
```diff
@@ -360,6 +361,13 @@ func (g *GlobalContext) ConfigureLogging() error {
 	style := g.Env.GetLogFormat()
 	debug := g.Env.GetDebug()
 	logFile := g.Env.GetLogFile()
+	if logFile == "" {
+		filePrefix := g.Env.GetLogPrefix()
+		if filePrefix != "" {
+			filePrefix = filePrefix + strings.Replace(time.Now().Format(time.RFC3339Nano), ":", "-", -1)
```
[the remark this answers is not in the document]
Comment (akalin-keybase) [reply go/libkb/globals.go]: ah, ok. Can you do `Format("20060102T150405.999999999Z0700")` then?

File: go/libkb/log_send.go [hunk 3]
```diff
@@ -480,6 +485,7 @@ func (l *LogSendContext) LogSend(statusJSON, feedback string, sendLogs bool, num
 		installLog = tail(l.G().Log, "install", logs.Install, numBytes)
 		systemLog = tail(l.G().Log, "system", logs.System, numBytes)
 		gitLog = tail(l.G().Log, "git", logs.Git, numBytes)
+		watchdogLog = tail(l.G().Log, "watchdog", logs.Watchdog, numBytes)
```
[the remark this answers is not in the document]
Comment (akalin-keybase) [reply go/libkb/log_send.go]: ah, cool, but itym https://github.com/keybase/keybase/pull/2424/files

**Comment by maxtaco** [comment]
Did something go wrong with the rebase? IT picked up a tons of commits from other people.

**Comment by zanderz** [comment]
whoops, let me try that again, sorry.

**Pull request merged**